ui/react-components/components/DatePicker/DateInput.jsx

```diff
@@ -28,36 +28,44 @@ const DateInput = (props) =>{
     };
 
     const inputRef = useRef(null);
-    const {value, onBlur, minDate, maxDate, isDisabled} = props;
+    const {value, onBlur, minDate, maxDate, isDisabled,hideDatePicker} = props;
     const [componentValue, setComponentValue] = useState(validateDate({dateValue: value, minDate,maxDate})? value:'');
     useEffect(() => {
         setComponentValue(validateDate({dateValue: value, minDate,maxDate})?value:'');
     }, [value, minDate]);
 
     const handleBlur=(e) =>{
         const inputValue = e.target.value;
-        const isValidDate = validateDate({dateValue: inputValue, minDate,maxDate});
-        if(isValidDate){
+        const isValidDate = validateDate({dateValue: inputValue, minDate, maxDate});
+        if (isValidDate) {
             onBlur(componentValue);
-        }else if(isNil(value)){
-            onBlur('')
+        } else if (isNil(value)) {
+            onBlur('');
             setComponentValue('')
-        }else {
-            setComponentValue(value)
+        } else {
+            setComponentValue(value);
+            inputRef.current.focus();
         }
     };
 
     const handleClear = (e) =>{
         if(!isDisabled) {
             setComponentValue('');
-            onBlur('')
+            onBlur('');
+            inputRef.current.focus();
         }
     };
 
     const handleKeyDown= (e) => {
         if (e.key === 'Enter') {
-           inputRef.current.blur();
+            inputRef.current.blur();
+            const inputValue = e.target.value;
+            const isValidDate = validateDate({dateValue: inputValue, minDate, maxDate});
+            if (isValidDate)
+                hideDatePicker && hideDatePicker();
```

Comment:
I think hideDatePicker does not make sense for all scenario's. Can we instead call it handleEnterKeyDown or something similar?

Reply:
CHanged them